raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -270,23 +293,20 @@ private void becomeUnattachedFollower(int epoch) throws IOException {
         boolean isLeader = quorum.isLeader();
         if (quorum.becomeUnattachedFollower(epoch)) {
             resetConnections();
-
-            if (isLeader) {
```

Comment:
Can you explain why this is no longer needed? Previously if we were leader, the timer would have been initialized to `Long.MAX_VALUE`, so we had to reset it here.